protostar/testing/environments/cairo_test_execution_environment.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+from typing import Any, Optional
+
+from starkware.cairo.common.cairo_function_runner import CairoFunctionRunner
+from starkware.cairo.lang.compiler.program import Program
+from starkware.cairo.lang.vm.vm_exceptions import VmException
+
+from protostar.testing.environments.cairo_test_cheatcode_factory import (
+    CairoTestCheatcodeFactory,
+)
+from protostar.testing.environments.test_execution_environment import (
+    TestExecutionEnvironment,
+    TestExecutionResult,
+)
+from protostar.testing.starkware.execution_resources_summary import (
+    ExecutionResourcesSummary,
+)
+from protostar.testing.test_environment_exceptions import RevertableException
+
+
+class CairoTestExecutionEnvironment(TestExecutionEnvironment):
+    def __init__(self, program: Program, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self._program = program
+        self._max_steps = self.state.config.max_steps
+        self._profiling = self.state.config.profiling
+
+    async def execute(self, function_name: str) -> TestExecutionResult:
+        assert (
+            not self.state.config.profiling
+        ), "Profiling is not supported yet"  # TODO: Plug in profiling
+
+        with self.state.output_recorder.redirect("test"):
+            return TestExecutionResult(
+                execution_resources=await self.execute_test_case(function_name)
+            )
+
+    # TODO: Extract execution resources from function run
+    # TODO: Estimate gas if self.state.config.gas_estimation_enabled
```

Comment:
Can we avoid "TODO/What" comments? I don't know "why" it was not implemented in the first place and if there's a task for it. If there's a task for it, I am not sure what's the point of this comment.

Reply:
I'm leaving those TODOs solely because of missing logic that should be plugged into the place that the TODO is placed (based on previous logic). I will link to the tasks in the next commit.

Reply:
I added issue numbers from the milestone to the TODOs